### DISCUSSION TITLE
이미지 업로드 시 이미지 관리 방법에 대한 논의

### DISCUSSION THREAD

**Discussion by gracefulBrown** [woowacourse/prolog#360] [Feature]
게시글에 이미지를 첨부할 경우 S3에 저장하게 하여 이미지를 노출하게 했을 때
기존 게시글이 삭제된 경우 이미지를 어떻게 처리할 것인지에 대한 논의가 필요

1. 게시글이 지워저도 이미지는 영원히 남아있도록 하기
2. 배치를 통해 주기적으로(1주일에 1번) 게시글이 지워졌는지 확인 후 이미지를 삭제 시키기

**Comment by sihyung92** [comment]
2번!!
저는 깃헙 이슈란을 이용해서 이미지를 업로드한뒤 해당 이미지 저장 링크를 활용해 블로깅을 하는 얍샙이거든요 그럴 때마다 얘넨 이미지 스토리지를 유지하기 위해 얼마나 많은 돈을 박고 있을까?? 그런데 내가 거기까지 걱정해줄건 아니지 알아서 하겠지 뭐 하는 쓸데없는 생각을 하곤 한답니다

깃헙은 마이크로소프트란 든든한 물주가 있지만 저희는 아직 없으니까요 출처가 사라진 이미지에 대해선 정리해서 이미지 스토리지 공간을 최대한 아끼는게 어떨까 싶네요

**Reply by seovalue** [reply]
저도 2번 동의해요오~~~
심지어 게시글이 지워질 때 바로 이미지를 삭제해도 괜찮다고 생각하는 입장으로서 1,2중엔 2번을 더 선호합니다!
전에 프로필 이미지 관련해서도 사용자가 프로필 이미지를 삭제하면 남아있어야할까?에 대해 고민했었는데 웨지 말대로 아직 한정된 공간이니 불필요한 이미지에 대해서는 처리를 하는 것이 좋을 것 같다구 생각함돠!
👍

**Comment by Hyeon9mak** [comment]
저도 2번 동의합니다. 제가 사용자라면 S3 안에 제가 지웠던 게시글의 이미지가 남아 있길 원하지 않을거 같아요.

**Comment by gracefulBrown** [comment]
2번으로 가시죠ㅎㅎ
